p2p/protocol/ping/ping_test.go

```diff
@@ -1,4 +1,4 @@
-package ping
+package ping_test
```

Comment:
Why is this change needed?

Reply:
we get an import cycle without it, because the test uses basic hosts.

Reply:
I will not that this is a good practice for testing a package from the perspective of a 'user' of a package as it only gives you access to the exported functionality of the `ping` package. Though it can make testing certain things more difficult especially in the domain of ipfs and libp2p.

Reply:
I will note, that this is a good practice for testing a package from the perspective of a 'user' of a package as it only gives you access to the exported functionality of the package. Though it can make testing certain things more difficult especially in the domain of ipfs and libp2p.